### PR TITLE
Use helper-builder-react-jsx inside plugin-transform-react-inline-elements

### LINKED ISSUE #6276: 7.0.0-beta transform-react-inline-elements broken for multiline string props
When using `transform-react-inline-elements` in the latest 7 betas, multiline string props don't transpile correctly. This leads to unterminated string constants which are not valid JavaScript.

### Input Code

```jsx
var test = <T default="
  some string
" />;
```

Currently this outputs the following:

```js
var test = _jsx(T, {
  "default": "
    some string
  "
});
```

which is not valid JS

### Babel Configuration

```js
{
  "presets": ["react"],
  "plugins": ["transform-react-inline-elements"]
}
```

### Expected Behavior
Valid JS is outputted. It should output the following given the input code:

```js
var test = _jsx(T, {
  "default": " some string "
});
```

### Current Behavior
Bad JS outputted. See input code section to see the input.

### Possible Solution
Not sure, but this seems to have broken from 6 -> 7, so it's likely something newer?

### Context
Attempted to update a relatively large internal project to 7 betas to see if there were any issues with it.

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.1
| node             | 6
| npm              | 3

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | Fixes #6276 
| Patch: Bug Fix?          | 👍
| Major: Breaking Change?  | 
| Minor: New Feature?      | 
| Tests Added/Pass?        | 👍
| Spec Compliancy?         | 
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  | 

This PR removes the duplicated implementation of the jsx transformer, so it makes it easier to avoid bugs.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Maybe we could also remove the spread element deoptimization, since `babel-helper-builder-react-jsx` handles it?

File: packages/babel-plugin-transform-react-inline-elements/test/fixtures/regressions/6276/actual.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+var test = <T default="
+    some string
```
Comment (Andarist) [inline packages/babel-plugin-transform-react-inline-elements/test/fixtures/regressions/6276/actual.js]: @nicolo-ribaudo isnt this SyntaxError? raw multiline strings are allowed only in template literals

Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-react-inline-elements/test/fixtures/regressions/6276/actual.js]: If I'm reading the JSX specification correctly, the only disallowed character is " (thus \n is allowed)
https://facebook.github.io/jsx/

Comment (Andarist) [reply packages/babel-plugin-transform-react-inline-elements/test/fixtures/regressions/6276/actual.js]: Indeed, it seems its allowed. Always thought that jsx is only a sugar for createElement calls and the rest is just JS. Dont like it doesnt conform to JS in those regards

Comment (hzoo) [reply packages/babel-plugin-transform-react-inline-elements/test/fixtures/regressions/6276/actual.js]: There is a lot of logic for strings in the transform at least

**Pull request merged**